src/python/bot/fuzzers/afl/engine.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""AFL engine interface."""
+
+from bot.fuzzers.afl import launcher
+from lib.clusterfuzz.fuzz import engine
+
+
+# TODO(mbarbella): Remove this check once this file is fully implemented.
+# pylint: disable=unused-variable
+class AFLEngine(engine.Engine):
+  """AFL engine implementation."""
+
+  @property
+  def name(self):
+    return 'afl'
+
+  def prepare(self, corpus_dir, target_path, build_dir):  # pylint: disable=unused-argument
+    """Prepare for a fuzzing session, by generating options. Returns a
+    FuzzOptions object.
+
+    Args:
+      corpus_dir: The main corpus directory.
+      target_path: Path to the target.
+      build_dir: Path to the build directory.
+
+    Returns:
+      A FuzzOptions object.
+    """
+    afl_config = launcher.AFLConfig.from_target_path(target_path)
+    arguments = afl_config.additional_afl_arguments
+    strategies = launcher.FuzzingStrategies(target_path).to_strategy_dict()
```

Comment:
currently i think mutations using radamsa etc are done in the fuzz() method of the runner. This should be moved to prepare() here as well -- maybe add a TODO?

Reply:
Good point. Added a TODO.